src/msix/pack/ContentType.cpp

```diff
@@ -0,0 +1,137 @@
+            { "xml",   { "text/xml", APPX_COMPRESSION_OPTION_NORMAL} },
+            { "xsd",   { "text/xml", APPX_COMPRESSION_OPTION_NORMAL} },
+            // Audio types
+            { "aiff",  { "audio/x-aiff", APPX_COMPRESSION_OPTION_NORMAL} },
+            { "au",    { "audio/basic", APPX_COMPRESSION_OPTION_NORMAL} },
+            { "m4a",   { "audio/mp4", APPX_COMPRESSION_OPTION_NONE} },
+            { "mid",   { "audio/mid", APPX_COMPRESSION_OPTION_NORMAL} },
+            { "mp3",   { "audio/mpeg", APPX_COMPRESSION_OPTION_NONE} },
+            { "smf",   { "audio/mid", APPX_COMPRESSION_OPTION_NORMAL} },
+            { "wav",   { "audio/wav", APPX_COMPRESSION_OPTION_NORMAL} },
+            { "wma",   { "audio/x-ms-wma", APPX_COMPRESSION_OPTION_NONE} },
+            // Image types
+            { "bmp",   { "image/bmp", APPX_COMPRESSION_OPTION_NORMAL} },
+            { "emf",   { "image/x-emf", APPX_COMPRESSION_OPTION_NORMAL} },
+            { "gif",   { "image/gif", APPX_COMPRESSION_OPTION_NONE} },
+            { "ico",   { "image/vnd.microsoft.icon", APPX_COMPRESSION_OPTION_NORMAL} },
+            { "jpg",   { "image/jpeg", APPX_COMPRESSION_OPTION_NONE} },
+            { "jpeg",  { "image/jpeg", APPX_COMPRESSION_OPTION_NONE} },
+            { "png",   { "image/png", APPX_COMPRESSION_OPTION_NONE} },
+            { "svg",   { "image/svg+xml", APPX_COMPRESSION_OPTION_NORMAL} },
+            { "tif",   { "image/tiff", APPX_COMPRESSION_OPTION_NORMAL} },
+            { "tiff",  { "image/tiff", APPX_COMPRESSION_OPTION_NORMAL} },
+            { "wmf",   { "image/x-wmf", APPX_COMPRESSION_OPTION_NORMAL} },
+            // Video types
+            { "avi",   { "video/avi", APPX_COMPRESSION_OPTION_NONE} },
+            { "mpeg",  { "video/mpeg", APPX_COMPRESSION_OPTION_NONE} },
+            { "mpg",   { "video/mpeg", APPX_COMPRESSION_OPTION_NONE} },
+            { "mov",   { "video/quicktime", APPX_COMPRESSION_OPTION_NONE} },
+            { "wmv",   { "video/x-ms-wmv", APPX_COMPRESSION_OPTION_NONE} }
+        };
+
+        auto findExt = extToContentType.find(ext);
+        if (findExt == extToContentType.end())
+        {
+            // if it's not in the map these are the defaults
+            return std::make_pair("application/octet-stream", APPX_COMPRESSION_OPTION_NORMAL);
+        }
+        return findExt->second;
+    }
+
+    const std::string ContentType::GetPayloadFileContentType(APPX_FOOTPRINT_FILE_TYPE footprintFile)
+    {
+        if (footprintFile == APPX_FOOTPRINT_FILE_TYPE_MANIFEST)
+        {
+            return "application/vnd.ms-appx.manifest+xml";
+        }
+        if (footprintFile == APPX_FOOTPRINT_FILE_TYPE_BLOCKMAP)
+        {
+            return "application/vnd.ms-appx.blockmap+xml";
+        }
+        if (footprintFile == APPX_FOOTPRINT_FILE_TYPE_SIGNATURE)
+        {
+            return "application/vnd.ms-appx.signature";
+        }
+        // TODO: add other ones if needed, otherwise throw
+        NOTIMPLEMENTED

```

Comment:
>NOTIMPLEMENTED [](start = 8, length = 14)

NOTSUPPORTED

Reply:
Although a more verbose exception would probably be better.

---
In reply to: [278648510](https://github.com/Microsoft/msix-packaging/pull/95#discussion_r278648510) [](ancestors = 278648510)

Reply:
Change to throw not support with a log message

---
In reply to: [278648651](https://github.com/Microsoft/msix-packaging/pull/95#discussion_r278648651) [](ancestors = 278648651,278648510)